src/flows/pipes/Columns/index.tsx

```diff
@@ -0,0 +1,57 @@
+import View, {TableColumnKey} from './view'
+
+export default register => {
+  register({
+    type: 'columnEditor',
+    family: 'passThrough',
+    component: View,
+    button: 'Column Editor',
+    initial: {
+      updatedTableKeys: {},
+    },
+    generateFlux: (pipe, create, append) => {
+      append(`__CURRENT_RESULT__ |> limit(n: 100)`)
+      if (!Object.values(pipe.updatedTableKeys).length) {
+        return
+      }
+
+      const toggleHide = Object.entries(
+        pipe.updatedTableKeys as {[_: string]: TableColumnKey}
+      ).reduce((a, [k, v]) => {
+        if (!v.visible) {
+          if (v.name !== k) {
+            a.push(`"${v.name}"`)
+          } else {
+            a.push(`"${k}"`)
+          }
+          return a
+        } else {
+          return a
+        }
+      }, [])
+
+      const rename = Object.entries(
+        pipe.updatedTableKeys as {[_: string]: TableColumnKey}
+      ).reduce((a, [k, v]) => {
+        if (k === v.name) {
+          return a
+        } else {
+          a.push(`${k}: "${v.name}"`)
+          return a
+        }
+      }, [])
+
+      if (rename.length) {
+        create(`
+                __PREVIOUS_RESULT__
+                |> rename(columns: {${rename.join(', ')}})
+                `)
+      }
+
+      if (toggleHide.length) {
+        create(`__PREVIOUS_RESULT__
+               |> drop(columns: [${toggleHide.join(', ')}])`)
+      }
```

Comment:
I think I'm a little confused as to what we're doing here. Is the goal of this to rename a column, then drop the column?

Reply:
No, its to be _able_ to do both. Thats why they are in separate conditionals. One for toggling visibility and one for changing name.

Reply:
so this will only let you either rename a column or drop them i think. i'd expect this to work better:
```
if (rename.length || toggleHide.length) {
    let query = '__PREVIOUS_RESULT__'
    if (rename.length) {
        query += ` |> rename(columns: {${rename.join(', ')}})`
    }
    if (toggleHide.length) {
        query += ` |> drop(columns: [${toggleHide.join(', ')}])`
    }
    create(query)
}
```

Reply:
and does the toggleHide part have to be `toggleHide.map(h => '"${h}"').join(', ')`?

Reply:
disregard the last part.. seems like you did that in the reduce

Reply:
Tested this and it allows you to do both